src/training_data.py

```diff
@@ -100,19 +100,19 @@ def load_luis_data(self,filename):
 
     def load_api_data(self, files):
         for filename in files:
-            data = json.loads(open(filename,'rb').read())
+            data = json.loads(codecs.open(filename, encoding='utf-8').read())
             # get only intents, skip the rest. The property name is the target class
             if "userSays" not in data:
                 continue
 
             intent = data.get("name")
             for s in data["userSays"]:
-                text = unicode("".join(map(lambda chunk: chunk["text"], s.get("data"))))
+                text = "".join(map(lambda chunk: chunk["text"], s.get("data")))
                 # add entities to each token, if available
                 entities = []
                 for e in filter(lambda chunk: "alias" in chunk or "meta" in chunk, s.get("data")):
-                    val = u"\s*" + unicode(e["text"])
-                    start = text.find(unicode(e["text"]))
+                    val = u"\s*" + e["text"]
```

Comment:
I've just spotted a mistake in `load_luis_data` which is now also here. `"value"` of the entity should be the matched part of the string. in `read_luis_data` we're actually constructing a regex and setting that. the `value` part of the entity should be `text[start:end]`. Could you please fix that for both?